plugins/woocommerce/includes/class-wc-mobile-messaging-handler.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * Prepares formatted mobile deep link navigation link for order mails
+ *
+ * @package WooCommerce\Emails
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Mobile_Messaging_Handler
+ */
+class WC_Mobile_Messaging_Handler {
+
+	const OPEN_ORDER_INTERVAL_DAYS = 30;
+
+	/**
+	 * Prepares mobile messaging with a deep link
+	 *
+	 * @param WC_Order $order order that mobile message is created for.
+	 * @param ?int     $blog_id  of blog to make a deep link for.
+	 * @param DateTime $now      current DateTime.
+	 *
+	 * @return string|null
```

Comment:
For consistency, what do you think of either following this convention (to match what we use a few lines above, in the case of `@param ?int $blog_id` and in the method signature itself)?

```suggestion
	 * @return ?string
```

Or indeed the reverse: use the `string|null` and `int|null` "union type" style within docblocks? ... Not vital we change this but it strikes me as being more readable when we are consistent.

Reply:
Yes, I completely agree. I've decided to stick with `?string` as it's more compact.